c7n/resources/s3.py

```diff
@@ -2621,8 +2621,116 @@ def process_bucket(self, bucket):
                 raise e
 
 
+class KMSKeyResolverMixin(object):
+    """Builds a dictionary of region specific ARNs"""
+
+    arns = dict()
```

Comment:
mutable class variables are problematic, this should be an instance attribute.

Reply:
ah interesting, it was not apparent that was how python would do that.  that would obviously be an issue